tensorpipe/channel/cuda_ipc/context_impl.cc

```diff
@@ -25,11 +26,46 @@ namespace cuda_ipc {
 
 namespace {
 
+std::string deviceUUID(int device) {
+  cudaDeviceProp props;
+  TP_CUDA_CHECK(cudaGetDeviceProperties(&props, device));
+
+  std::ostringstream uuid;
+  uuid << std::hex;
+
+#if CUDART_VERSION > 9020
+  uuid << std::setfill('0') << std::setw(2);
+  for (int j = 0; j < 16; ++j) {
+    uuid << (props.uuid.bytes[j] & 0xff);
+  }
+#else // CUDART_VERSION <= 9020
+  // CUDA <=9.2 does not have a uuid device property. Falling back to using PCI
+  // ids.
```

Comment:
The CUDA driver seems to have a function for that though also in version 9.2.

https://docs.nvidia.com/cuda/archive/9.2/cuda-driver-api/group__CUDA__DEVICE.html#group__CUDA__DEVICE_1g987b46b884c101ed5be414ab4d9e60e4